targets/frontend/src/pages/storage/[container]/index.js

```diff
@@ -0,0 +1,327 @@
+/** @jsx jsx */
+
+import { useRouter } from "next/router";
+import prettyBytes from "pretty-bytes";
+import { useEffect, useState } from "react";
+import {
+  IoIosTrash,
+  IoMdCheckmark,
+  IoMdClipboard,
+  IoMdCloseCircleOutline,
+  IoMdDownload,
+} from "react-icons/io";
+import { Button, IconButton } from "src/components/button";
+import { Layout } from "src/components/layout/auth.layout";
+import { DropZone } from "src/components/storage/DropZone";
+import { withCustomUrqlClient } from "src/hoc/CustomUrqlClient";
+import { withUserProvider } from "src/hoc/UserProvider";
+import { useDebouncedState } from "src/hooks/";
+import { deleteFile, listFiles, uploadFiles } from "src/lib/azure";
+import { timeSince } from "src/lib/duration";
+import useSWR, { mutate } from "swr";
+import {
+  Card,
+  Field,
+  Flex,
+  jsx,
+  Label,
+  Message,
+  Select,
+  Spinner,
+} from "theme-ui";
+
+const onDeleteClick = function (container, file) {
+  const confirmed = confirm(
+    `Êtes-vous sûr(e) de vouloir définitivement supprimer ${file.name} ?`
+  );
+  if (confirmed) {
+    deleteFile(container, file.name)
+      .then(() => {
+        mutate("files");
+      })
+      .catch(() => {
+        alert("Impossible de supprimer le fichier :/");
+      });
+  }
+};
+
+function FilesPage() {
+  const router = useRouter();
+  const container = router.query.container;
+  const { data = [], error } = useSWR("files", listFiles(container));
+
+  const [hasClipboardApi, setHasClipboardApi] = useState(false);
+  const [currentClip, setCurrentClip] = useState("");
+  const [isSearching, setSearching] = useState(false);
+  const [search, setSearch] = useState("");
+  const [sort, setSort] = useState("mostRecent");
+  const [filter, setFilter] = useState("");
+  const [
+    displayedFiles,
+    setDisplayedFiles,
+    setDebouncedDisplayedFiles,
+  ] = useDebouncedState([], 400);
```

Comment:
why use a useDebouncedState ? swr trigger render when it needs to ?

Reply:
It's not about SWR it's about delagging the input when typing. The display rerender is longer than the typing, we don't want the typing to "freeze"

Reply:
as I understand, I would use debouncedState on the filter input and then update the files accordingly

Reply:
Nice ! Way better thanks